gql_code_builder/lib/src/operation/data.dart

```diff
@@ -242,21 +261,36 @@ ListBuilder<Field> _buildFields() => ListBuilder<Field>(
       ],
     );
 
-ListBuilder<Constructor> _buildConstructors() => ListBuilder<Constructor>(
+ListBuilder<Constructor> _buildConstructors(
+  String name,
+  List<InlineFragmentNode> inlineFragments,
+) =>
+    ListBuilder<Constructor>(
       <Constructor>[
         Constructor(
           (b) => b
-            ..requiredParameters = ListBuilder<Parameter>(
-              <Parameter>[
-                Parameter(
-                  (b) => b
-                    ..name = "data"
-                    ..toThis = true,
-                ),
-              ],
-            )
+            ..name = inlineFragments.isEmpty ? null : "fromData"
+            ..requiredParameters.add(Parameter(
+              (b) => b
+                ..name = "data"
+                ..toThis = true,
+            ))
             ..constant = true,
         ),
+        if (inlineFragments.isNotEmpty)
+          Constructor(
+            (b) => b
+              ..factory = true
+              ..requiredParameters.add(Parameter((b) => b..name = "data"))
+              ..body = Code([
```

Comment:
I couldn't figure out how to build a switch statement with `code_builder` (as far as I can tell from the source, switch statements are not supported), so I had to write the code out manually.

Reply:
np